CMakeLists.txt

```diff
@@ -1976,6 +1976,11 @@ if(CONFIG_MINIMAL_LIBCPP)
   zephyr_compile_options($<$<COMPILE_LANGUAGE:CXX>:$<TARGET_PROPERTY:compiler-cpp,nostdincxx>>)
 endif()
 
+if(CONFIG_LLVM_PROFILE)
+  zephyr_compile_options(-fprofile-instr-generate -fcoverage-mapping)
+  zephyr_link_libraries(-fprofile-instr-generate -fcoverage-mapping)
```

Comment:
One of the cmake experts will need to tell me if this is the right spot for this or not.  It seems weird to have all the compiler arguments here at the top level, but putting it into the cmake/toolchain/ tree didn't work because the default target there is "zephyr_interface" (or something to that effect).

Reply:
can you elaborate, what did not work / how did it not work ?

zephyr_compile_options and zephyr_link_libraries both works on the Zephyr interface, so there should not be any difference there.

Reply:
(If you do what I commented in the other comment, redefining $<TARGET_PROPERTY:{compiler/linker},coverage>, you should not need to do this change.)

But if you want to go this way, I'd expect you could put it around https://github.com/zephyrproject-rtos/zephyr/blob/main/arch/posix/CMakeLists.txt#L108  without problems(?)

And if you do so, please add also the equivalent 
```
target_compile_options(native_simulator INTERFACE ..)
target_link_options(native_simulator INTERFACE ..)
```
so it also works for native_sim